src/NServiceBus.Persistence.AzureStorage/Timeout/TimeoutLogic/TimeoutPersister.cs

```diff
@@ -400,27 +396,23 @@ string Sanitize(string s)
             return results.SafeFirstOrDefault();
         }
 
-        static Task UpdateSuccessfulRead(CloudTable table, TableOperation operation)
+        static async Task UpdateSuccessfulRead(CloudTable table, TableOperation operation)
```

Comment:
@SimonCropp what was the reason for this change?

Reply:
https://particularsoftware.slack.com/archives/C1607JM6G/p1504173950000203

Reply:
I see. If fails the tests now. Reason: the logic has to slightly change because we no longer ignore just `HttpStatusCode.PreconditionFailed` and `HttpStatusCode.Conflict` scenarios. We also need to take into consideration when update is successful and no content is coming back (`HttpStatusCode.NoContent`).

Reply:
I also suggest to use `HttpStatusCode` and not the numbers. Easier to read and understand what each of those codes stands for. /cc @yvesgoeleven